core/clients/persistient_storage/classes/s3.py

```diff
@@ -0,0 +1,30 @@
+import os
+
+from core.clients.persistient_storage.classes.main import PersistientStorageClient
+
+class S3Client(PersistientStorageClient):
+    PREFIX = "s3://"
+
+    def __init__(self, path: str):
+        bucket, *prefix = path.removeprefix(self.PREFIX).split("/")
+
+        self.bucket = bucket
+        self.prefix = "/".join(prefix)
+
+        import boto3
```

Comment:
** Note: I addressed https://github.com/yeti-platform/yeti/pull/1137#discussion_r1776114928 before this one. `boto3` has already been made optional.

I was importing it like this because for most they will not be using `boto3` but just on startup if I added it to the top it would always be included and loaded. Since boto3 is a decently large package, figured we shouldn't import it unless we actually need it

Curious for your preference for implementation here; I am imagining three possibilities based on what you said:

Option A: Always include S3Client and alert when used but `boto3` not imported
```py
try:
	import boto3
except ImportError:
	boto3 = None
	logging.warn('boto3 is not imported, if you wish to use s3 file storage please install with `poetry install --extras s3`')

class S3Client(FileStorageClient):
    def __init__(self, path: str):
		if boto3 is None:
			logging.want("Attempting to use `S3Client` without `boto3` installed; install with `poetry install --extras s3`"
			raise ImportError("boto3 is not installed")
		...
```

Option B: Only include S3Client when `boto3` exists
```py
try:
	import boto3
except ImportError:
	boto3 = None
	logging.warn('boto3 is not imported, if you wish to use s3 file storage please install with `poetry install --extras s3`')

if boto3 is not None:
	class S3Client(FileStorageClient):
	    ...
```

Reply:
The first implementation looks great! The error is clear and the warnings should guide the user to install things properly.

Reply:
Added in [8d8edaa](https://github.com/yeti-platform/yeti/pull/1137/commits/8d8edaa0cdc34542244b27893ec051b349e05878)